backend/src/test/java/com/google/sps/EmailNotifierTest.java

```diff
@@ -0,0 +1,177 @@
+      throws MessagingException, IOException, GeneralSecurityException {
+    ArgumentCaptor<Message> argument = ArgumentCaptor.forClass(Message.class);
+    emailNotifier.notifyUser();
+    verify(messages).send(any(), argument.capture());
+    String subjectName = convertToMimeMessage(argument.getValue()).getSubject();
+    Truth.assertWithMessage("These Two strings should be the same ")
+        .that("Ad-Lib Meeting Found")
+        .isEqualTo(subjectName);
+  }
+
+  @Test
+  public void testMessageShouldHaveCorrectBodyText()
+      throws MessagingException, IOException, GeneralSecurityException {
+    ArgumentCaptor<Message> argument = ArgumentCaptor.forClass(Message.class);
+    emailNotifier.notifyUser();
+    verify(messages).send(any(), argument.capture());
+    String bodyText = convertToMimeMessage(argument.getValue()).getContent().toString();
+    Truth.assertWithMessage("These Two strings should be the same ")
+        .that(
+            " Hey John Please Join your Ad-Lib meeting via the link below : \n"
+                + " http://meet.google.com/new")
+        .isEqualTo(bodyText);
+  }
+
+  @Test
+  public void testMessageShouldHaveIncorrectBodyText()
+      throws MessagingException, IOException, GeneralSecurityException {
+    ArgumentCaptor<Message> argument = ArgumentCaptor.forClass(Message.class);
+    emailNotifier.notifyUser();
+    verify(messages).send(eq("me"), argument.capture());
+    String realString = convertToMimeMessage(argument.getValue()).getContent().toString();
+    Truth.assertWithMessage("These Two strings should not be the same ")
+        .that("anyString()")
+        .isNotEqualTo(realString);
+  }
+
+  @Test
+  public void testMessageShouldHaveCorrectRecipients()
+      throws MessagingException, IOException, GeneralSecurityException {
+    ArgumentCaptor<Message> argument = ArgumentCaptor.forClass(Message.class);
+    emailNotifier.notifyUser();
+    verify(messages).send(any(), argument.capture());
+    Address[] allRecipients = convertToMimeMessage(argument.getValue()).getAllRecipients();
+    Address[] correctRecipients =
+        new Address[] {
+          new InternetAddress("jdoe@gmail.com "),
+        };
+    Truth.assertWithMessage("These Two arrays should be the same ")
+        .that(correctRecipients)
+        .isEqualTo(allRecipients);
+    Truth.assertThat(allRecipients[0]).isEqualTo(correctRecipients[0]);
+  }
+
+  @Test
+  public void testMessageShouldHaveIncorrectRecipients()
+      throws MessagingException, IOException, GeneralSecurityException {
+    ArgumentCaptor<Message> argument = ArgumentCaptor.forClass(Message.class);
+    emailNotifier.notifyUser();
+    verify(messages).send(any(), argument.capture());
+    Address[] allRecipients = convertToMimeMessage(argument.getValue()).getAllRecipients();
+    Truth.assertWithMessage("These Two arrays should be the same ")
+        .that(new InternetAddress[] {new InternetAddress()})
+        .isNotEqualTo(allRecipients);
+  }
```

Comment:
Can you elaborate on the scenario this test is testing? The test above checks that in this scenario allRecipients will not be empty right?

Reply:
The test above this one asserted that the given Attendees were correct, given that the current implementation is only sending an email to one person thus would need two function calls in the future this test asserts that given that the attendees cannot be empty given the constraints of the functions being called I am comparing the mock with behavior that is never expected to be present

Reply:
Does that that clear some of your confusion ?

Reply:
Attendees == recipients?

But is it possible for this test to fail when the other test doesn't fail? The one above has the same setup and action but asserts that the recipients has a single entry. That would also seem to cover this test method's assertion that the recipients are not empty.

Reply:
Yes I meant recipients instead of attendees. After your explanation i can see where the test case is redundant. Acknowledged